### PR TITLE
Adds safe_move_directory to fs_utils

### PR DESCRIPTION
**Pull request by arnavgarg1**
Adds a companion method `safe_move_directory` to fs_utils which is inspired by `safe_move_file`.

### CODE REVIEW

**Review by justinxzhao — APPROVED** [review]

File: ludwig/utils/fs_utils.py [hunk 1]
```diff
@@ -182,6 +182,26 @@ def safe_move_file(src, dst):
...
+            # random UUID so if different processes are copying into `<dst>`, they don't overlap in their tmp copies.
+            copy_id = uuid.uuid4()
+            tmp_dst = f"{dst}.{copy_id}.tmp"
+            shutil.copytree(src, tmp_dst)
+
+            # Atomic replace file onto the new name, and clean up original source file.
+            os.replace(tmp_dst, dst)
+            os.unlink(src)
+        else:
+            raise
```
Comment (justinxzhao) [inline ludwig/utils/fs_utils.py:202]: Is this right?

Comment (arnavgarg1) [reply ludwig/utils/fs_utils.py:202]: @justinxzhao It seems to work as expected for me. Just fixed, I noticed that my commit changing this to `shutil.copytree` from `shutil.copyfile` didn't get pushed up

File: ludwig/utils/fs_utils.py [hunk 2]
```diff
@@ -182,6 +182,26 @@ def safe_move_file(src, dst):
             raise
 
 
+@DeveloperAPI
+def safe_move_directory(src, dst):
```
Comment (tgaddair) [inline ludwig/utils/fs_utils.py:186]: Test

Comment (arnavgarg1) [reply ludwig/utils/fs_utils.py:186]: Adding it right now

**Review by tgaddair — APPROVED** [review]

**Pull request merged**